rawspec_file.c

```diff
@@ -73,8 +96,26 @@ void * dump_file_thread_func(void *arg)
 {
   callback_data_t * cb_data = (callback_data_t *)arg;
 
-  if(cb_data->fd && cb_data->h_pwrbuf){
-    if(cb_data->per_ant_out){
+  // Single antenna case
+  if(cb_data->fd && cb_data->h_pwrbuf && (cb_data->Nant == 1)) {
+      if(cb_data->debug_callback)
+        printf("dump_file_thread_func: write for nants=0\n");
+      if(cb_data->flag_fbh5_output) {
+        fbh5_write(&(cb_data->fbh5_ctx_ant[0]),
+                   &(cb_data->fb_hdr), 
+                   cb_data->h_pwrbuf, 
+                   cb_data->h_pwrbuf_size,
+                   cb_data->debug_callback);
+      } else {
+        write(cb_data->fd[0], 
+              cb_data->h_pwrbuf, 
+              cb_data->h_pwrbuf_size);
+      } // if(cb_data->flag_fbh5_output) 
+  }
+
+  // Multiple antennas, split output
+  if(cb_data->fd && cb_data->h_pwrbuf) {
```

Comment:
Could we have this consume the prior if to make the impossibility of `per_at_out && Nants>1` clearer, and doubly guarded agains:
```
  if(cb_data->fd && cb_data->h_pwrbuf) {
    if(cb_data->Nants == 1) {
    ...
    } else // if(cb_data->per_ant_out)
    {
    ...
    }
  }
```

Reply:
How about this .....

Reply:
Sure. I still like a hard 'else if' which then starts leading back to my initial thought.. But this is clearer too. It starts to nag at the idea of removing the `per_ant_output` flag, seeing as we always have a per_ant_output... I can tackle that in a following PR.

Reply:
It made the 3 cases easier to read and maintain for me.
That's why I restructured like that.